### PR TITLE
Fix prefilling non-default lang product name

### LINKED ISSUE #28530: Multilanguages -Product V2- Empty fields when another language is selected lead to an exception in the payment step
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

When adding the name of a new product in the page product V2 we have the following problem

1. possibility to save the name even with Empty fields when another language is selected
2. exception in the payment step
3. the size of the product in the FO is not correct

https://user-images.githubusercontent.com/89604350/169096887-a6de332d-12ca-4b24-b10a-48e74c2ac287.mp4

image

image

### Expected behavior

_No response_

### Steps to reproduce

1. Go to BO > International> Translation> Add mor than a language 
2. Go to the page Products (experimental)>click on New product on experimental page> Add new product
3. choose a name when english is selected 
4. set a pricing
5. set a quantity
6. Set the product to online
7. Save and pulish
8. change the language of the name and delete the name in the other language (keep the name empty for other language)
9. Only the name for default language is filled
10. Save
11. Go to FO change the language(different from default)
12. See errors:

- Product is without name

- the size of the product is not as default

13. try to pass an order see exception in the payment step

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

8.0

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | refer to https://github.com/PrestaShop/PrestaShop/issues/28530. Now when it should be impossible to have empty product name in any language (its prefilled with default language name)
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Closes https://github.com/PrestaShop/PrestaShop/issues/28530
| Related PRs       | 
| How to test?      | refer to https://github.com/PrestaShop/PrestaShop/issues/28530 (NOTE - the bug only occured when only the non-default name value was emptied, but default name was not changed in same request)
| Possible impacts? |

### CODE REVIEW

File: src/Adapter/Product/CommandHandler/UpdateProductBasicInformationHandler.php [hunk 1]
```diff
@@ -110,7 +110,7 @@ private function fillUpdatableProperties(
             // Go through all the product languages and make sure name is filled for each of them
             $productLanguages = array_keys($product->name);
             foreach ($productLanguages as $languageId) {
-                if (empty($product->name[$languageId]) && empty($localizedNames[$languageId])) {
```
Comment (zuk3975) [inline src/Adapter/Product/CommandHandler/UpdateProductBasicInformationHandler.php:113]: this statement failed therefor not setting the default name when product name existed before, but was being removed in this request. (`empty($product->name[$languageId])` was falsy)

File: src/Adapter/Product/CommandHandler/UpdateProductBasicInformationHandler.php [hunk 2]
```diff
@@ -110,7 +110,7 @@ private function fillUpdatableProperties(
...
+                if (empty($localizedNames[$languageId])) {
```
Comment (zuk3975) [inline src/Adapter/Product/CommandHandler/UpdateProductBasicInformationHandler.php:113]: now we always just check if its trying to empty the name (if yes, we use default, no matter if product had the name before or not)

**Review by matthieu-rolland — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @zuk3975

**Comment by sallemiines** [comment]
I will check this PR

**Comment by sallemiines** [comment]
Hello @zuk3975 

The issue is solved
- I run the automatic tests :**OK**
- Add more than one language in the international -> translation page
- Create new product name with default language in V2 product page
- Check that the name is displayed in the list of languages
- Select another language and set the product name empty, then save
- Verify that the product name is auto-saved with the existing value in the default language.
- Go to FO, and verify that the product name exists with any language chosen
- Pass an order successfully
- Disable a language, and check that in the product page that the disabled language dosen't  exist (the issue already exists ->  [issues/16626](https://github.com/PrestaShop/PrestaShop/issues/16626))
- Add empty product fields and set it offline
- Add empty product fields and set it online
- Add products name of different types: standard, pack, variations, combination


https://user-images.githubusercontent.com/31663369/188604882-96f60580-1e93-41a0-9c70-aea9909867e0.mp4

Thank you !

**Pull request merged**